tests/request.rs

```diff
@@ -0,0 +1,36 @@
+use nats::MessageError;
+use async_std::task;
+use std::str::from_utf8;
+use std::sync::mpsc;
+
+#[test]
+fn no_responders() -> Result<(), String> {
+    let nc = nats::Options::new().connect("demo.nats.io").unwrap();
+    let res = nc.request("noresponders", "message").unwrap();
+    match res.err {
```

Comment:
I think that this has to be part of Result as it's idiomatic approach.
This is very ugly solution. `Message.err` exists only to pass the error between dispatcher and subscribers.
Though before implementation, I would like to have an agreement that we want to change signature of the request `Result` as it is breaking change, unfortunately.